File: tmp/test-tailwind-fix/ontology.config.ts
import { defineOntology, userContext } from 'ont-run';
import { z } from 'zod';

// Import resolver functions - TypeScript enforces return types match outputs
import healthCheck from './resolvers/healthCheck.js';
import getUser from './resolvers/getUser.js';
import deleteUser from './resolvers/deleteUser.js';

export default defineOntology({
  name: 'my-api',

  environments: {
    dev: { debug: true },
    prod: { debug: false },
  },

  // Pluggable auth - customize this for your use case
  // Return { groups, user } for row-level access control
  auth: async (req) => {
    const token = req.headers.get('Authorization');
    // Return access groups and optional user data
    // This is where you'd verify JWTs, API keys, etc.
    if (!token) return { groups: ['public'] };
    if (token === 'admin-secret') {
      return {
        groups: ['admin', 'support', 'public'],
        user: { id: 'admin-1', email: 'admin@example.com' },
      };
    }
    return {
      groups: ['support', 'public'],
      user: { id: 'user-1', email: 'user@example.com' },
    };
  },

  accessGroups: {
    public: { description: 'Unauthenticated users' },
    support: { description: 'Support agents' },
    admin: { description: 'Administrators' },
  },

  entities: {
    User: { description: 'A user account' },
  },

  functions: {
    // Example: Public function
    healthCheck: {
      description: 'Check API health status',
      access: ['public', 'support', 'admin'],
      entities: [],
      inputs: z.object({}),
      resolver: healthCheck,
    },

    // Example: Restricted function with row-level access
    getUser: {
      description: 'Get user details by ID',
      access: ['support', 'admin'],
      entities: ['User'],
      inputs: z.object({
        userId: z.string().uuid(),
        // currentUser is injected from auth - not visible to API callers
        currentUser: userContext(z.object({
          id: z.string(),
          email: z.string(),
        })),
      }),
      resolver: getUser,
    },

    // Example: Admin-only function
    deleteUser: {
      description: 'Delete a user account',
      access: ['admin'],
      entities: ['User'],
      inputs: z.object({
        userId: z.string().uuid(),
        reason: z.string().optional(),
      }),
      resolver: deleteUser,
    },
  },
});

File: examples/basic/ontology.config.go
package main

import (
	ont "github.com/vanna-ai/ont-run/pkg/ontology"
	"github.com/vanna-ai/ont-run/examples/basic/resolvers"
)

func DefineOntology() *ont.Config {
	return &ont.Config{
		Name:         "basic-app",
		Title:        "Basic App - Example Ontology",
		Instructions: "A simple example app with health check and user management. Use 'healthCheck' to verify the server is running, 'getUser' to fetch a user by ID, and 'listUsers' to paginate through all users.",
		UUID:         "example-basic-app-uuid", // Replace with a real UUID in production
		Cloud:        true,                     // Enable cloud registration with ont-run.com

		AccessGroups: map[string]ont.AccessGroup{
			"public": {
				Description: "Unauthenticated users",
			},
			"admin": {
				Description: "Administrators with full access",
			},
		},

		Entities: map[string]ont.Entity{
			"User": {
				Description: "A user account in the system",
			},
		},

		Functions: map[string]ont.Function{
			"healthCheck": {
				Description: "Check if the server is running",
				Access:      []string{"public"},
				Entities:    []string{},
				Inputs:      ont.Object(map[string]ont.Schema{}),
				Outputs: ont.Object(map[string]ont.Schema{
					"status":    ont.String(),
					"timestamp": ont.String().DateTime(),
				}),
				Resolver: resolvers.HealthCheck,
			},
			"getUser": {
				Description: "Get user by ID",
				Access:      []string{"admin"},
				Entities:    []string{"User"},
				Inputs: ont.Object(map[string]ont.Schema{
					"id": ont.String().UUID(),
				}),
				Outputs: ont.Object(map[string]ont.Schema{
					"id":    ont.String(),
					"name":  ont.String(),
					"email": ont.String().Email(),
					"tags":  ont.Array(ont.String()), // Array - never nil
				}),
				Resolver: resolvers.GetUser,
			},
			"listUsers": {
				Description: "List all users with pagination",
				Access:      []string{"admin"},
				Entities:    []string{"User"},
				Inputs: ont.Object(map[string]ont.Schema{
					"page":     ont.Integer().Min(1),
					"pageSize": ont.Integer().Min(1).Max(100),
				}),
				Outputs: ont.Object(map[string]ont.Schema{
					"users": ont.Array(ont.Object(map[string]ont.Schema{
						"id":    ont.String(),
						"name":  ont.String(),
						"email": ont.String().Email(),
					})),
					"total": ont.Integer().NonNegative(),
					"page":  ont.Integer(),
				}),
				Resolver: resolvers.ListUsers,
			},
		},
	}
}
